vclib-openid/src/commonMain/kotlin/at/asitplus/wallet/lib/oidc/OidcSiopVerifier.kt

```diff
@@ -170,42 +172,26 @@ class OidcSiopVerifier(
         responseMode: String? = null,
         state: String? = uuid4().toString(),
     ): AuthenticationRequestParameters {
+        val containerJwt = FormatContainerJwt(algorithms = arrayOf(JwsAlgorithm.ES256.text))
         val vpFormats = FormatHolder(
-            msoMdoc = if (credentialScheme?.credentialFormat == ConstantIndex.CredentialFormat.ISO_18013)
-                FormatContainerJwt(algorithms = arrayOf(JwsAlgorithm.ES256.text)) else null,
-            jwtVp = FormatContainerJwt(algorithms = arrayOf(JwsAlgorithm.ES256.text)),
+            msoMdoc = if (credentialRepresentation == ConstantIndex.CredentialRepresentation.ISO_MDOC) containerJwt else null,
+            jwtVp = containerJwt,
         )
         val metadata = RelyingPartyMetadata(
             redirectUris = arrayOf(relyingPartyUrl),
             jsonWebKeySet = JsonWebKeySet(arrayOf(agentPublicKey.toJsonWebKey())),
             subjectSyntaxTypesSupported = arrayOf(URN_TYPE_JWK_THUMBPRINT, PREFIX_DID_KEY),
             vpFormats = vpFormats,
         )
-        val mainConstraint = when (credentialScheme?.credentialFormat) {
-            ConstantIndex.CredentialFormat.W3C_VC -> ConstraintField(
-                path = arrayOf("$.type"),
-                filter = ConstraintFilter(
-                    type = "string",
-                    pattern = credentialScheme.vcType,
-                )
-            )
-
-            ConstantIndex.CredentialFormat.ISO_18013 -> ConstraintField(
-                path = arrayOf("$.mdoc.doctype"),
-                filter = ConstraintFilter(
-                    type = "string",
-                    pattern = NAMESPACE_MDL,
-                )
-            )
-
-            else -> ConstraintField(
-                path = arrayOf("$.type"),
-                filter = ConstraintFilter(
-                    type = "string",
-                    pattern = "AnyCredential",
-                )
-            )
+        val typeConstraint = credentialScheme?.let {
+            when (credentialRepresentation) {
+                ConstantIndex.CredentialRepresentation.PLAIN_JWT -> it.vcConstraint()
```

Comment:
We have both functions for all credentials, so we can encode mDL as W3C VCs for debugging, right?!

Reply:
Yes, we can